src/time/gadgets.h

```diff
@@ -6,27 +6,83 @@
  * Please see the COPYING and CONTRIBUTORS files for details.
  */
 
-/* DEBUG: section 21    Time Functions */
+#ifndef SQUID__SRC_TIME_GADGETS_H
+#define SQUID__SRC_TIME_GADGETS_H
 
-#ifndef   SQUID_TIME_H
-#define   SQUID_TIME_H
-
-#include "rfc1123.h"
-
-#include <ctime>
-#include <iosfwd>
-/* NP: sys/time.h is provided by libcompat */
+/* old Squid API for for milli- and micro- second time access */
 
 /* Use uint64_t to store milliseconds */
 typedef uint64_t time_msec_t;
 
-/* globals for accessing time */
-extern struct timeval current_time;
+/// the current UNIX time in seconds (with microsecond decimal)
 extern double current_dtime;
+
+/* old C time_t API */
+
+#include <ctime>
+
+/// the current UNIX time in seconds
 extern time_t squid_curtime;
 
-time_t getCurrentTime(void);
-int tvSubMsec(struct timeval, struct timeval);
+/// Update squid_curtime (also current_time, and current_dtime)
+/// \returns new value of squid_curtime
+time_t getCurrentTime();
+
+namespace Time {
+
+/// Convert from ISO 3307 style time: YYYYMMDDHHMMSS or YYYYMMDDHHMMSS.xxx
+time_t ParseIso3307(const char *);
+
+/** Display time as a formatted human-readable string.
+ * Time string syntax used is from RFC 1123
+ * "www, DD MMM YYYY hh:mm:ss GMT"
+ *
+ * Output is only valid until next call to this function.
+ */
+const char *FormatRfc1123(time_t);
+
+/// Convert from RFC 1123 style time: "www, DD MMM YYYY hh:mm:ss ZZZ"
+time_t ParseRfc1123(const char *);
+
+/** Display time as a formatted human-readable string.
+ * Time syntax is
+ * "YYYY/MM/DD hh:mm:ss"
+ *
+ * Output is only valid until next call to this function.
+ */
+const char *FormatStrf(time_t);
+
+/** Display time as a formatted human-readable string.
+ * Time string syntax used is that of Apache httpd.
+ * "DD/MMM/YYYY:hh:mm:ss zzzz"
+ *
+ * Output is only valid until next call to this function.
+*/
+const char *FormatHttpd(time_t);
+
+} // namespace Time
+
+/* old C struct timeval API */
+
+#include <iosfwd>
+// NP: sys/time.h is included by libcompat
```

Comment:
I recommend removing this comment. This header should not know what inside `squid.h` provides `sys/time.h`, and anybody curious about it can figure it out.

If you keep it, please either keep the original comment (so I do not need to review/adjust it) or drop the `NP: ` prefix. As we have discussed, it is not a standard/common-enough abbreviation that many readers are likely to recognize, FWIW, I still do not know what it is despite trying to find out!

Reply:
> I recommend removing this comment. This header should not know what inside `squid.h` provides `sys/time.h`, and anybody curious about it can figure it out.
> 

Copy-n-paste from the old includes list. But fine, done in [82347a0](https://github.com/squid-cache/squid/pull/1001/commits/82347a06be8f3cd9f24ab99432d7713d3fd8c87e).

> If you keep it, please either keep the original comment (so I do not need to review/adjust it) or drop the `NP: ` prefix. As we have discussed, it is not a standard/common-enough abbreviation that many readers are likely to recognize, FWIW, I still do not know what it is despite trying to find out!

Interesting, it is so common over here I had not considered it as being completely unknown anywhere else. With a colon it is "Note Please:", Australian English form of the Latin phrase "Nota Bene".

Reply:
We have discussed these `NP` problems before. In most Squid contexts (including this one), any such prefix is essentially noise, but if you are sure it is needed in any specific context, please use its more common, better understood, and less ambiguous `N.B.` variant.